src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -136,12 +136,19 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
     # Checking whether optional extra values file has been provided.
     values_file_provided, values_file = utils.get_values_file()
 
-    # Validate the helm environment file for Dogfood.
-    dp_endpoint_dogfood = None
-    release_train_dogfood = None
+    release_train_custom = None
+    metadata = utils.get_metadata(cmd.cli_ctx.cloud.endpoints.resource_manager)
+    # Read and validate the helm values file for Dogfood.
+    config_dp_endpoint = None
     if cmd.cli_ctx.cloud.endpoints.resource_manager == consts.Dogfood_RMEndpoint:
         azure_cloud = consts.Azure_DogfoodCloudName
-        dp_endpoint_dogfood, release_train_dogfood = validate_env_file_dogfood(values_file, values_file_provided)
+        config_dp_endpoint, release_train_custom = validate_env_file_dogfood(values_file, values_file_provided)
+    # Get the values or endpoints required for retreiving the Helm registry URL.
+    if "dataplaneEndpoints" in metadata:
```

Comment:
For Context, we added dataplane endpoints in ARM metadata response for the api version 2022-09-01 with this PR
https://msazure.visualstudio.com/DefaultCollection/One/_git/AzureUX-ARM/pullrequest/6838164 (internal)

Reply:
The dataplaneEndpoints values are not populated for the azure clouds at the moment, however we initialize it in our configs for the disconnected scenarios so that the correct endpoint values are returened.
In future Azure clouds could make use of these fields as well